### PR TITLE
feat: add func of getting specified type from workspace genericConfig

### PR DESCRIPTION
**Pull request by healthjyk**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

add func of getting specified type from workspace genericConfig.

### CODE REVIEW

File: pkg/workspace/util.go [hunk 1]
```diff
@@ -237,3 +211,63 @@ func CompleteWholeS3Config(config *workspace.S3Config) {
 		config.Region = region
 	}
 }
+
+// GetIntFieldFromGenericConfig returns the value of the key in config which should be of type int.
+// If exist but not int, return error. If not exist, return 0, nil.
+func GetIntFieldFromGenericConfig(config workspace.GenericConfig, key string) (int, error) {
```
Comment (adohe) [inline pkg/workspace/util.go:217]: GetIntFieldFromGenericConfig -> GetIntFromGenericConfig

Comment (adohe) [inline pkg/workspace/util.go:217]: compared to return error, return yes or no might be better?

File: pkg/workspace/util.go [hunk 2]
```diff
@@ -237,3 +211,63 @@ func CompleteWholeS3Config(config *workspace.S3Config) {
...
+	value, ok := config[key]
+	if !ok {
+		return 0, nil
+	}
+	i, ok := value.(int)
```
Comment (adohe) [inline pkg/workspace/util.go:222]: how about using `int32` here?

File: pkg/workspace/util.go [hunk 3]
```diff
@@ -237,3 +211,63 @@ func CompleteWholeS3Config(config *workspace.S3Config) {
...
+	if !ok {
+		return 0, fmt.Errorf("the value of %s is not map", key)
+	}
+	return i, nil
+}
+
+// GetStringFieldFromGenericConfig returns the value of the key in config which should be of type string.
+// If exist but not string, return error; If not exist, return "", nil.
+func GetStringFieldFromGenericConfig(config workspace.GenericConfig, key string) (string, error) {
```
Comment (adohe) [inline pkg/workspace/util.go:231]: GetStringFieldFromGenericConfig -> GetStringFromGenericConfig

File: pkg/workspace/util.go [hunk 4]
```diff
@@ -237,3 +211,63 @@ func CompleteWholeS3Config(config *workspace.S3Config) {
...
+	s, ok := value.(string)
+	if !ok {
+		return "", fmt.Errorf("the value of %s is not string", key)
+	}
+	return s, nil
+}
+
+// GetMapFieldFromGenericConfig returns the value of the key in config which should be of type map[string]any.
+// If exist but not map[string]any, return error; If not exist, return nil, nil.
+func GetMapFieldFromGenericConfig(config workspace.GenericConfig, key string) (map[string]any, error) {
```
Comment (adohe) [inline pkg/workspace/util.go:245]: GetMapFieldFromGenericConfig -> GetMapFromGenericConfig

File: pkg/workspace/util.go [hunk 5]
```diff
@@ -237,3 +211,63 @@ func CompleteWholeS3Config(config *workspace.S3Config) {
...
+	m, ok := value.(map[string]any)
+	if !ok {
+		return nil, fmt.Errorf("the value of %s is not map", key)
+	}
+	return m, nil
+}
+
+// GetStringMapFieldFromGenericConfig returns the value of the key in config which should be of type map[string]string.
+// If exist but not map[string]string, return error; If not exist, return nil, nil.
+func GetStringMapFieldFromGenericConfig(config workspace.GenericConfig, key string) (map[string]string, error) {
```
Comment (adohe) [inline pkg/workspace/util.go:259]: GetStringMapFieldFromGenericConfig -> GetStringMapFromGenericConfig

File: pkg/workspace/util.go (hunk 2 above)
Comment (healthjyk) [reply pkg/workspace/util.go:222]: If return `int32`, there should be a new func named `GetInt32FromGenericConfig` cause the returned type is fixed.

File: pkg/workspace/util.go (hunk 1 above)
Comment (healthjyk) [reply pkg/workspace/util.go:217]: If return yes or no, when using these functions, the calling one always still need to build an error. Besides, when calling `GetMapFromGenericConfig`, there are two kinds of errors, the first is `not map`, the second is `field is not string`, using `bool` is not enough.

Comment (healthjyk) [reply pkg/workspace/util.go:217]: fixed

File: pkg/workspace/util.go (hunk 3 above)
Comment (healthjyk) [reply pkg/workspace/util.go]: fixed

File: pkg/workspace/util.go (hunk 4 above)
Comment (healthjyk) [reply pkg/workspace/util.go]: fixed

File: pkg/workspace/util.go (hunk 5 above)
Comment (healthjyk) [reply pkg/workspace/util.go]: fixed

**Review by liu-hm19 — APPROVED** [review]
lgtm

**Pull request merged**